node/impl/full/eth.go

```diff
@@ -2075,25 +2071,34 @@ func newEthTxReceipt(ctx context.Context, tx ethtypes.EthTx, replay *api.InvocRe
 		LogsBloom:        ethtypes.EmptyEthBloom[:],
 	}
 
-	if replay.MsgRct.ExitCode.IsSuccess() {
+	if lookup.Receipt.ExitCode.IsSuccess() {
 		receipt.Status = 1
-	}
-	if replay.MsgRct.ExitCode.IsError() {
+	} else {
 		receipt.Status = 0
 	}
 
-	receipt.GasUsed = ethtypes.EthUint64(replay.MsgRct.GasUsed)
+	receipt.GasUsed = ethtypes.EthUint64(lookup.Receipt.GasUsed)
 
 	// TODO: handle CumulativeGasUsed
 	receipt.CumulativeGasUsed = ethtypes.EmptyEthInt
 
-	effectiveGasPrice := big.Div(replay.GasCost.TotalCost, big.NewInt(replay.MsgRct.GasUsed))
+	// TODO: avoid loading the tipset twice (once here, once in the when we convert the message to a txn)
```

Comment:
Not critical right now (it's guaranteed to be cached). It's just a bit annoying.

Reply:
Agreed, but grammar of the comment is wonky (`once in the when we convert`).